config.yml

```diff
@@ -1,21 +1,134 @@
 configuration_file_version: 1
 
-Comment: default values might be a bit crazy.. They are intended for testing. Change them.
+debug: true
 
+# Default values might be a bit crazy.. They are intended for testing. Change them.
 
-Comment: kick_thresholds field ignores non-numerical characters and only reads three numbers separated, 
-Comment: non-numericals are there only to make configuration file more readable.
-Comment: Use 0 as an amount of seconds when you don't want to kick people.
+# kick_thresholds field ignores non-numerical characters and only reads three numbers separated, 
+#   non-numericals are there only to make configuration file more readable.
+#   Use 0 as an amount of seconds when you don't want to kick people.
 
 kick_thresholds:
-  - When 81 to 100 players online kick after 60 seconds
-  - When 51 to 80  players online kick after 1000 seconds
-  - When 2  to 50  players online kick after 20 seconds
-  - When 1  to 1  players online kick after 40 seconds  
-  
-  
-Comment: warnings filed reads the first number, the rest is text to be sent to a player
+  - When 81 to 500 players online kick after 60 seconds
+  - When 51 to 80  players online kick after 120 seconds
+  - When 2  to 50  players online kick after 240 seconds
+  - When 1  to 1  players online kick after 3600 seconds  
   
+bot_detector:
+  # Once the TPS get below acceptable_TPS, the BotDetector starts culling potential lag causer.
+  acceptable_TPS: 17.5
+  # critical_TPS_Change is the minimal change in TPS that needs to happen after kicking a player, 
+  #   so he is marked as someone causing lag and his name is logged.
+  critical_TPS_Change: 0.5
+  # kicking_frequency is how often a player gets kicked and the TPS change monitored afterwards 
+  #   in ticks
+  kicking_frequency: 1200
+  # Whether players get banned for a long time/ until the tick is high again
+  long_bans: false
+  # Length of long ban
+  ban_length: 21600000
+  # Max locations per player to store while measuring movement patterns
+  max_locations: 5
+  # Max suspects to consider per Detector round
+  max_suspects: 10
+  # Baseline movement (in MC blocks) to be considered for bot detection
+  min_baseline: 512
+  # Maximum scan rounds to "ignore" a player who was previously cleared as a lag source
+  max_reprieve: 20
+  # Scan radius in chunks around location, to check for lag causes
+  scan_radius: 2
+  # Configure the threshold and individual contribution of bound types
+  bounds:
+    relaxation_factor: 1.05
+    threshold: 2.5
+    contained: 1.5
+    contained_exclude_y: 1.5
+    volume_similar: 1.0
+    surface_similar: 1.0
+    nearly_contained: 0.75
+    nearly_contained_exclude_y: 0.75
+
+lag_scanner:
+  # Length of time to cache individual chunk results
+  cache_timeout: 900000

```

Comment:
You should probably add that this is in ms, because some other timings are in ticks. Also I think we can safely increase this to half an hour, it's almost impossible to cause bigger differences without creative mode in less than that.